addons/account/models/chart_template.py

```diff
@@ -311,6 +311,19 @@ def tax_template_changed(tax, template):
                 elif model_name == 'account.tax':
                     # Only update the tags of existing taxes
                     if xmlid not in xmlid2tax or tax_template_changed(xmlid2tax[xmlid], values):
+                        account_tax = self.ref(xmlid, raise_if_not_found=False)
+                        if not account_tax:
+                            existing_account_tax = self.env['account.tax'].search([
+                                *self.env['account.tax']._check_company_domain(company),
+                                ('name', '=', values.get('name')),
+                                ('type_tax_use', '=', values.get('type_tax_use'))
+                            ])
+                            if existing_account_tax:
+                                self.env['ir.model.data']._update_xmlids([{
+                                    'xml_id': f"account.{company.id}_{xmlid}",
+                                    'record': existing_account_tax,
+                                    'noupdate': True,
+                                }])
                         if self._context.get('force_new_tax_active'):
                             values['active'] = True
                         if xmlid in xmlid2tax:
```

Comment:
So basically I propose to remove all your code above and add this below (not tested, just to give you an idea):
```suggestion
                        if xmlid in xmlid2tax:
                            obsolete_xmlid.add(xmlid)
                            oldtax = xmlid2tax[xmlid]
                        else:
                            oldtax = current_taxes.filtered(lambda t:
                                t.name == values.get('name') and
                                t.type_tax_use == values.get('type_tax_use') and
                                t.tax_scope == values.get('tax_scope')
                            )
                        if unique_tax_name_key(oldtax) in unique_tax_name_keys:
                            oldtax.name = f"[old] {oldtax.name}"
```